core/src/main/scala/org/elasticmq/actor/QueueManagerActor.scala

```diff
@@ -27,15 +27,17 @@ class QueueManagerActor(nowProvider: NowProvider, limits: Limits) extends Replyi
             case Left(error) =>
               Left(QueueCreationError(queueData.name, error))
             case Right(_) =>
-              val actor = createQueueActor(nowProvider, queueData)
+              val actor = createQueueActor(nowProvider, queueData, queueMetadataListener)
               queues(queueData.name) = actor
+              queueMetadataListener.foreach(_ ! PersistQueue(queueData))
```

Comment:
since it's a listener, maybe the naming could go along these lines: `QueueCreated`, `QueueDeleted`? This would also indicate at which point the callback is done (*after* the fact, not before)